### PR TITLE
test: ts esnext

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

test

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
NA

**Summary**
Add test for ts esm

**Does this PR introduce a breaking change?**
no

**Other information**
Fix https://github.com/webpack/webpack-cli/issues/2458

### CODE REVIEW

File: packages/webpack-cli/lib/utils/prompt.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            rl.close();
+        });
+    });
+    rl.on('close', function () {
+        console.log('\nBYE BYE !!!');
+        process.exit(0);
+    });
+};
+
+module.exports = confirm;
```
[the remark this answers is not in the document]
Comment (anshumanv) [reply packages/webpack-cli/lib/utils/prompt.js]: eeks new file 😄  removed 👍

**Comment by anshumanv** [comment]
need to skip test for node < 14

**Comment by alexander-akait** [comment]
Yep

**Comment by anshumanv** [comment]
hmm timing out here but seems to pass locally

File: test/build/config-format/typescript-esnext/typescript.test.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// eslint-disable-next-line node/no-unpublished-require
+const { run } = require('../../../utils/test-utils');
+const { existsSync } = require('fs');
+const { resolve } = require('path');
+
+describe('webpack cli', () => {
+    it('should support typescript esnext file', async () => {
+        if (process.version.slice(1, 3) < 14) {
```
Comment (evenstensberg) [inline test/build/config-format/typescript-esnext/typescript.test.js:8]: version in its own variable

Comment (anshumanv) [reply test/build/config-format/typescript-esnext/typescript.test.js]: done

**Review by alexander-akait — DISMISSED** [review]

**Review by snitin315 — APPROVED** [review]

**Review by evenstensberg — DISMISSED** [review]

**Comment by anshumanv** [comment]
hm timing out on node 14 on macos 😞

**Comment by alexander-akait** [comment]
Very weird, can you investigate? Maybe we really need increase timeout here

**Comment by anshumanv** [comment]
Yep lets try increasing timeout, seems to work for me quickly in MacOS

File: test/build/config-format/typescript-esnext/typescript.test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    it('should support typescript esnext file', async () => {
+        const isMacOS = process.platform === 'darwin';
+        const majorNodeVersion = process.version.slice(1, 3);
+        if (majorNodeVersion < 14) {
+            expect(true).toBe(true);
+
+            return;
+        }
+
+        if (isMacOS && !isWebpack5) {
```
Comment (alexander-akait) [inline test/build/config-format/typescript-esnext/typescript.test.js:16]: Why here `isWebpack5`? Happens only for v4 on macos?

Comment (anshumanv) [reply test/build/config-format/typescript-esnext/typescript.test.js:16]: it's `!isWebpack5` i.e 4 😄

Comment (anshumanv) [reply test/build/config-format/typescript-esnext/typescript.test.js:16]: yes

Comment (anshumanv) [reply test/build/config-format/typescript-esnext/typescript.test.js:16]: which is very weird as I don't know how webpack version change is causing this

Comment (anshumanv) [reply test/build/config-format/typescript-esnext/typescript.test.js:16]: Doesn't work locally either 😞

Comment (alexander-akait) [reply test/build/config-format/typescript-esnext/typescript.test.js:16]: > Doesn't work locally either disappointed

Do you mean it freezes on macos locally too?

File: test/build/config-format/typescript-esnext/typescript.test.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            expect(true).toBe(true);
+
+            return;
+        }
+
+        const { exitCode, stderr, stdout } = await run(__dirname, ['-c', './webpack.config.ts'], {
+            nodeOptions: ['--loader=ts-node/esm'],
+            env: { WEBPACK_CLI_FORCE_LOAD_ESM_CONFIG: true },
```
Comment (alexander-akait) [inline test/build/config-format/typescript-esnext/typescript.test.js:24]: We should avoid using `WEBPACK_CLI_FORCE_LOAD_ESM_CONFIG` here, it should works without it, locally fine

Comment (anshumanv) [reply test/build/config-format/typescript-esnext/typescript.test.js:24]: yep lets avoid it

Comment (anshumanv) [reply test/build/config-format/typescript-esnext/typescript.test.js]: done

File: test/build/config-format/typescript-esnext/typescript.test.js (hunk 3 above)
Comment (anshumanv) [reply test/build/config-format/typescript-esnext/typescript.test.js:16]: yep

Comment (alexander-akait) [reply test/build/config-format/typescript-esnext/typescript.test.js:16]: Very weird, bug we will drop webpack v4 in near future, so let's merge

**Review by alexander-akait — APPROVED** [review]

**Pull request merged**